pandas/core/generic.py

```diff
@@ -4574,50 +4573,61 @@ def pipe(self, func, *args, **kwargs):
     """)
 
     _shared_docs['transform'] = ("""
-    Call function producing a like-indexed %(klass)s
-    and return a %(klass)s with the transformed values
+    Call ``func`` on self producing a %(klass)s with transformed values
+    and that has the same axis length as self.
 
     .. versionadded:: 0.20.0
 
     Parameters
     ----------
-    func : callable, string, dictionary, or list of string/callables
-        To apply to column
+    func : function, string, list of functions and/or strings or dict
+        Function to use for transforming the data. If a function, must either
+        work when passed a %(klass)s or when passed to %(klass)s.apply.
 
-        Accepted Combinations are:
+        Accepted combinations are:
 
         - string function name
         - function
-        - list of functions
-        - dict of column names -> functions (or list of functions)
+        - list of functions and/or function names
+        - dict of axis labels -> functions, function names or list of such.
+    %(axis)s
+    *args
+        Positional arguments to pass to `func`.
+    **kwargs
+        Keyword arguments to pass to `func`.
```

Comment:
If you prefer, numpydoc also accepts having both in one line:
```
*args, **kwargs
    Arguments to pass to `func`.
```

Reply:
Alright, I did that, though my own preference would be to not use star arguments and use ``args=None, kwargs=None`` instead to better have a distinction what gets passed to ``func`` and what doesn't get passed on. But that's for a whole another discussion, and may be too late now :-)